x/valset/keeper/keep_alive.go

```diff
@@ -141,7 +141,11 @@ func (k Keeper) UpdateGracePeriod(ctx context.Context) {
 	}
 
 	vals := slice.Map(k.GetUnjailedValidators(ctx), func(i stakingtypes.ValidatorI) []byte {
-		return []byte(i.GetOperator())
+		bz, err := k.staking.ValidatorAddressCodec().StringToBytes(i.GetMoniker())
+		if err != nil {
+			panic(err)
```

Comment:
Don't panic in keeper production code.

Reply:
Do i need to print the error?

Reply:
Please don't resolve discussions before they're cleared up.

We don't want to use `panic` in production code when we don't need to. The idiomatic way of handling this is by changing the function signature to return an error and having calling code handle the error accordingly, e.g. by logging.

Curious though. Can we not just say `i.GetOperator().Bytes()` here?

Reply:
> Curious though. Can we not just say `i.GetOperator().Bytes()` here?

No, `i.GetOperator()` returns `string`.

Reply:
In that case, `[]byte(i.GetOperator())` should be fine.

Reply:
> In that case, `[]byte(i.GetOperator())` should be fine.

No, this will convert `bech32` address to bytes. In v0.47, `GetOperator()` returned `hex` addresses. However, in the latest release, it provides `bech32` addresses.

Reply:
Ugh, ugly. Okay, back to the error handling we are.